source/actions3/actions/AMDetectorSetDarkCurrentTimeActionInfo.cpp

```diff
@@ -0,0 +1,30 @@
+#include "AMDetectorSetDarkCurrentTimeActionInfo.h"
+
+AMDetectorSetDarkCurrentTimeActionInfo::AMDetectorSetDarkCurrentTimeActionInfo(double secondsDwell, const AMDetectorInfo &detectorInfo, QObject *parent) :
+	AMActionInfo3(QString(), QString(), QString(), parent)
+{
+	detectorInfo_.setValuesFrom(detectorInfo);
+	darkCurrentTime_ = secondsDwell;
+
+	setShortDescription(typeDescription());
+	setLongDescription(typeDescription());
+}
+
+AMDetectorSetDarkCurrentTimeActionInfo::AMDetectorSetDarkCurrentTimeActionInfo(const AMDetectorSetDarkCurrentTimeActionInfo &other) :
+	AMActionInfo3(other)
+{
+	detectorInfo_.setValuesFrom( *(other.detectorInfo()) );
+	darkCurrentTime_ = other.darkCurrentTime_;
```

Comment:
There might be a weird copy constructor rule that allows you to do this, but generally this should be ```darkCurrentTime_ = other.darkCurrentTime();``` because ```darkCurrentTime_``` is protected.